rust: Support showing running config only

cathay4t opened the pull request:
Support showing running config only excluding:
 * IP address retrieved by DHCP or IPv6 auto configuration.
 * DNS client resolver retrieved by DHCP or IPv6 auto configuration.
 * Routes retrieved by DHCPv4 or IPv6 router advertisement.
 * LLDP neighbor information.

API:
 * Rust: `NetworkState.set_running_config_only(bool)`.
 * Python: `libnmstate.show_running_config()`.

Integration test cases enabled.

cathay4t committed:
rust cli: Support showing in JSON format

Add support of `show --json` argument.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
rust: Remove leftover debug println

Remove the debug println from rust/src/lib/ifaces/inter_ifaces.rs.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (changes requested):
CI is failing for nmstatectl rust tests, could you check them? Thanks!

cathay4t committed:
rust: Support showing running config only

Support showing running config only excluding:
 * IP address retrieved by DHCP or IPv6 auto configuration.
 * DNS client resolver retrieved by DHCP or IPv6 auto configuration.
 * Routes retrieved by DHCPv4 or IPv6 router advertisement.
 * LLDP neighbor information.

API:
 * Rust: `NetworkState.set_running_config_only(bool)`.
 * Python: `libnmstate.show_running_config()`.

Integration test cases enabled.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
CI fixed.

cathay4t commented:
This has wait long enough. Force merging.
